src/shims/fs.rs

```diff
@@ -630,3 +630,64 @@ impl FileMetadata {
         Ok(Some(FileMetadata { mode, size, created, accessed, modified }))
     }
 }
+
+fn stat_write_buf<'tcx, 'mir>(
```

Comment:
This is specifically the macOS stat layout, right? If yes, the method name should reflect that.